COMFIE/App/Router/Router.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  Router.swift
+//  COMFIE
+//
+//  Created by Anjin on 3/6/25.
+//
+
+import Combine
+import SwiftUI
+
+@Observable
+class Router {
+    // 네비게이션 경로
+    var path: [Route] = .init()
+    
+    // 앱 실행 시, 2초간 로딩 화면 종료 여부
+    var isLoadingViewFinished: Bool = false
+    
+    // 앱 최초 실행 여부 - 온보딩 화면 전환 확인
+    var hasEverOnboarded: Bool = false {
+        didSet {
+            UserDefaultsService().save(value: hasEverOnboarded, key: .hasEverOnboarded)
+        }
+    }
+    
+    init() {
+        let userDefaults = UserDefaultsService()
+        let result = userDefaults.load(type: Bool.self, key: .hasEverOnboarded)
+        
+        switch result {
+        case .success(let data):
+            hasEverOnboarded = data
+        case .failure:
+            hasEverOnboarded = false
+        }
+    }
+    
+    // 가장 상위 View - 앱 진입 시 파악
+    @ViewBuilder func rootView(_ router: Router) -> some View {
+        if isLoadingViewFinished == false {
+            Route.loading.view(router)     // 앱 진입 시, 로딩 View
+        } else if hasEverOnboarded == false {
+            Route.onboarding.view(router)  // 앱 최초 실행 O -> 온보딩 View
+        } else {
+            Route.memo.view(router)        // 앱 최초 실행 X -> 메모 View
+        }
+    }
```

Comment:
위 코드를 봤을 때 하나 의문이었던 것은 Router 클래스 내에 만들어진 함수인데 매개변수로 또 Router를 갖고있다는 점이었어요! 
그래서 어떻게 해결할 수 있을까 하다가 gpt 의 도움을 빌려보았습니다.

제가 들었던 의문을 `self`를 활용해서 해결할 수 있더라고요! 

```suggestion
    @ViewBuilder func rootView() -> some View {
        if isLoadingViewFinished == false {
            Route.loading.view(self)     // 앱 진입 시, 로딩 View
        } else if hasEverOnboarded == false {
            Route.onboarding.view(self)  // 앱 최초 실행 O -> 온보딩 View
        } else {
            Route.memo.view(self)        // 앱 최초 실행 X -> 메모 View
        }
    }
```